SIT.Manager.Avalonia/Services/AppUpdaterService.cs

```diff
@@ -0,0 +1,181 @@
+{
+    private const string MANAGER_VERSION_URL = @"https://api.github.com/repos/stayintarkov/SIT.Manager.Avalonia/releases/latest";
+    private const string SITMANAGER_PROC_NAME = "SIT.Manager.Avalonia.Desktop.exe";
+    private const string SITMANAGER_WIN_RELEASE_URL = @"https://github.com/stayintarkov/SIT.Manager.Avalonia/releases/latest/download/win-x64.zip";
+    private const string SITMANAGER_LINUX_RELEASE_URL = @"https://github.com/stayintarkov/SIT.Manager.Avalonia/releases/latest/download/linux-x64.tar";
+
+    private readonly ILogger<AppUpdaterService> _logger = logger;
+    private readonly HttpClient _httpClient = httpClient;
+    private readonly IManagerConfigService _managerConfigService = managerConfigService;
+
+    private static string ReleaseUrl
+    {
+        get
+        {
+            if (OperatingSystem.IsWindows())
+            {
+                return SITMANAGER_WIN_RELEASE_URL;
+            }
+            else if (OperatingSystem.IsLinux())
+            {
+                return SITMANAGER_LINUX_RELEASE_URL;
+            }
+            throw new NotImplementedException("No Release URL found for this platform");
+        }
+    }
+
+    private static async Task MoveManager(DirectoryInfo source, string destination)
+    {
+        await MoveManager(source, new DirectoryInfo(destination));
+    }
+
+    private static async Task MoveManager(DirectoryInfo source, DirectoryInfo destination)
+    {
+        IEnumerable<DirectoryInfo> directories = source.EnumerateDirectories();
+        IEnumerable<FileInfo> files = source.EnumerateFiles();
+        destination.Create();
+
+        foreach (DirectoryInfo directory in directories)
+        {
+            if (directory.Name.Equals("backup", StringComparison.InvariantCultureIgnoreCase))
+            {
+                continue;
+            }
+            DirectoryInfo newDestination = destination.CreateSubdirectory(directory.Name);
+            await MoveManager(directory, newDestination);
+        }
+
+        foreach (FileInfo file in files)
+        {
+            try
+            {
+                file.MoveTo(Path.Combine(destination.FullName, file.Name));
+            }
+            catch (Exception) { }
+        }
+    }
+
+    public async Task<bool> CheckForUpdate()
+    {
+        if (_managerConfigService.Config.LookForUpdates)
+        {
+            try
+            {
+                Version currentVersion = Assembly.GetExecutingAssembly().GetName().Version ?? new Version("0");
+
+                string versionJsonString = await _httpClient.GetStringAsync(MANAGER_VERSION_URL);
+                GithubRelease? latestRelease = JsonSerializer.Deserialize<GithubRelease>(versionJsonString);
+                if (latestRelease != null)
+                {
+                    Version latestVersion = new(latestRelease.name);
+                    return latestVersion.CompareTo(currentVersion) > 0;
```

Comment:
I know this is back from the old manager. Can't we replace this with `latestVersion > currentVersion`?

Reply:
Good shout lemme change that